src/Tickets/Emails/Dispatcher.php

```diff
@@ -0,0 +1,513 @@
+		 * @param Email_Abstract $this    The email object.
+		 */
+		$headers = apply_filters( "tec_tickets_emails_dispatcher_{$email_slug}_headers", $headers, $this );
+
+		// Enforce as an array
+		return (array) $headers;
+	}
+
+	/**
+	 * Get the formatted headers for this dispatcher.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	public function get_headers_formatted(): array {
+		$original_headers = $this->get_headers();
+		$headers          = [];
+
+		foreach ( $original_headers as $key => $value ) {
+			$headers[] = "{$key}: {$value}";
+		}
+
+		return $headers;
+	}
+
+	/**
+	 * Get email attachments.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	public function get_attachments(): array {
+		/**
+		 * Filter the attachments.
+		 *
+		 * @since TBD
+		 *
+		 * @param array          $attachments The attachments.
+		 * @param Email_Abstract $this        The email object.
+		 */
+		$attachments = apply_filters( 'tec_tickets_emails_dispatcher_attachments', $this->attachments, $this );
+
+		$email_slug = $this->get_email()->slug;
+
+		/**
+		 * Filter the attachments for the particular email.
+		 *
+		 * @since TBD
+		 *
+		 * @param array          $attachments The attachments.
+		 * @param Email_Abstract $this        The email object.
+		 */
+		$attachments = apply_filters( "tec_tickets_emails_dispatcher_{$email_slug}_attachments", $attachments, $this );
+
+		return (array) $attachments;
+	}
+
+	/**
+	 * Get email recipient.
+	 *
+	 * @since TBD
+	 *
+	 * @return ?string The email recipient.
+	 */
+	public function get_to(): ?string {
+		/**
+		 * Allow filtering the email recipient globally.
+		 *
+		 * @since TBD
+		 *
+		 * @param string         $to   The email recipient.
+		 * @param Email_Abstract $this The email object.
+		 */
+		$to = apply_filters( 'tec_tickets_emails_dispatcher_to', $this->to, $this );
+
+		$email_slug = $this->get_email()->slug;
+
+		/**
+		 * Allow filtering the email recipient for the particular email.
+		 *
+		 * @since TBD
+		 *
+		 * @param string         $to   The email recipient.
+		 * @param Email_Abstract $this The email object.
+		 */
+		$to = apply_filters( "tec_tickets_emails_dispatcher_{$email_slug}_to", $to, $this );
+
+		return null === $to ? $to : (string) $to;
+	}
+
+	/**
+	 * Sets where we will dispatch to.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $value
+	 */
+	public function set_to( string $value ): void {
+		$this->to = $value;
+	}
+
+	/**
+	 * Get the subject of the email.
+	 *
+	 * @since TBD
+	 *
+	 * @return ?string
+	 */
+	public function get_subject(): ?string {
+		/**
+		 * Allow filtering the email subject globally.
+		 *
+		 * @since TBD
+		 *
+		 * @param string         $subject  The email subject.
+		 * @param string         $id       The email id.
+		 * @param string         $template Template name.
+		 * @param Email_Abstract $this     The email object.
+		 */
+		$subject = apply_filters( 'tec_tickets_emails_dispatcher_subject', $this->subject, $this );
```

Comment:
Is the docblock above correct? I don't see `$id` or `$template`.

Reply:
Thank you!